pkg/cosign/mock.go

```diff
@@ -53,17 +54,31 @@ func (m *mock) getSignatures(signedImgRef name.Reference) ([]oci.Signature, bool
 
 	sigs := make([]oci.Signature, 0, len(results))
 	for _, sp := range results {
-		sigs = append(sigs, &sig{cosignPayload: sp})
+		ociSig, err := GetOCISignatureFromCosignSignedPayload(sp)
+		if err != nil {
+			return nil, false, err
+		}
+		sigs = append(sigs, ociSig)
 	}
 
 	return sigs, true, nil
 }
 
-type sig struct {
-	oci.Signature
-	cosignPayload cosign.SignedPayload
-}
-
-func (s *sig) Payload() ([]byte, error) {
-	return s.cosignPayload.Payload, nil
+func GetOCISignatureFromCosignSignedPayload(sp cosign.SignedPayload) (oci.Signature, error) {
```

Comment:
Shorten `GetOCISignatureFromCosignSignedPayload` to `GetSignatureFromSignedPayload` or just `GetSignature`.

No need to repeat package names. Also the params indicate we are passing a signed payload and getting back a signature, so the "From..." is optional.